util/signing/hashicorp-vault/src/cryptography/ed25519/mod.rs

```diff
@@ -0,0 +1,28 @@
+use aws_sdk_kms::types::{KeySpec, KeyUsageType, SigningAlgorithmSpec};
+use signer::cryptography::secp256k1::Secp256k1;
+
+/// Defines the needed methods for providing a definition of cryptography used with AWS KMS
+pub trait AwsKmsCryptography {
+	/// Returns the [KeySpec] for the desired cryptography
+	fn key_spec() -> KeySpec;
+
+	/// Returns the [KeyUsageType] for the desired cryptography
+	fn key_usage_type() -> KeyUsageType;
+
+	/// Returns the [SigningAlgorithmSpec] for the desired cryptography
+	fn signing_algorithm_spec() -> SigningAlgorithmSpec;
+}
```

Comment:
I'm not sure a trait is the right way to define data. A struct is easier and do the same. Each Kms provide its own filed struct.
And why it's define for each curve?

Reply:
```rust
pub struct Spec {
    key_spec: KeySpec,
    key_usage_type: KeyUsageType,
    signing_algorithm_spec: SigningAlgorithmSpec
}

pub trait AwsKmsCryptography {
	/// Returns the [Spec] for the desired cryptography
	fn spec() -> Spec;
}
```

...is also fine by me. But, I think you need to use the trait system in some capacity to do the sort of compositional approach suggested.

Reply:
@musitdev @mzabaluev Do we want to shift the specs to something like this?

Reply:
Yes but it's a AWS specific spec? Do we want to define this in the application API?

Reply:
For all I'm concerned, `AwsKmsCryptography` could be a private helper trait in aws-kms (but not in this crate, this discussion is on top of a misplaced duplicate), and so its precise details don't matter much.